WPMedia/fusion-news-theme-blocks
# Pen 1378 video promo block

## Description
- update customfields for video promo blocks
- add inherit global content check box
- update README

## Jira Ticket
- [PEN-1378](https://arcpublishing.atlassian.net/browse/PEN-1378)

## Acceptance Criteria
- [ ] The group name Configure Video should be renamed to Configure Content (because we want to use this name consistently across all blocks so users always know where to configure content)

- [ ] An additional checkbox Custom Field should be added to the Configure Content group called Inherit global content
When this is checked true, the block should inherit the page’s global content

- [ ] The field Video UUID should be removed from the block completely (because this is duplicative – the video’s ID can be entered in the content source ID field)

- [ ] The Ratio custom field should be removed from the block completely (there’s no need for users to change video ratio) 

- [ ] The Autoplay custom field should be moved to a new group called Video settings

- [ ] The group named Content should be renamed Display settings

## Test Steps
- on the theme repo, update `.env` file:
  - link your local engine-theme-sdk, e.g. `ENGINE_SDK_REPO={PATH/TO/LOCAL/engine-theme-sdk}`
  - add `videoOrg=prod`
  - add `videoOrg=corecomponents` 
  - go to environment/index.json file and add `"videoEnv": "prod",`

- under block repo
  - create a new page and add video promo block
  
--- 
Test against content configurations:
- [ ] make sure inherit global content checkbox is UNCHECKED 
- [ ] update the `website_url` with `/video/2020/09/21/a-shallow-river-streaming-through-a-bed-of-rocks/` under customfields
- [ ] click the update button
- [ ] confirm the video match with https://corecomponents.arcpublishing.com/videocenter/video/v2/#/clip/5dc1d1c252faff00098c3399



--- 
Test against global content
- [ ] update global content on the page with `website_url` with `/video/2019/12/17/a-snowy-waterfall/`
<img width="1023" alt="Screen Shot 2020-11-19 at 3 48 10 PM" src="https://user-images.githubusercontent.com/4411687/99742033-7497f780-2a87-11eb-94ae-1f0957c4940a.png">
- [ ] confirm the videomatch with https://corecomponents.arcpublishing.com/videocenter/video/v2/#/clip/5df2b5164cedfd00092d093c

---
Test Inherit Global content
- [ ] once the content config and global content is set up as above
- [ ] click the Inherit Global Content button and it should toggle between above videos

--- 
Test Display Settings
- [ ] add "title" under "Display setting
- [ ] add "description" under "display setting"
- [ ] both title and description will display on screen


## Effect Of Changes
### Before
- customfields include ratio, video uuid
<img width="884" alt="Screen Shot 2020-11-05 at 2 54 58 PM" src="https://user-images.githubusercontent.com/4411687/98305539-e6941b00-1f76-11eb-87de-2bed9dd74cba.png">

### After
<img width="904" alt="Screen Shot 2020-11-19 at 5 36 34 PM" src="https://user-images.githubusercontent.com/4411687/99745145-ed9a4d80-2a8d-11eb-8e02-4bb2a24f58d5.png">


 

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
